plugin/builtin/sortordertransformer/SortOrderTransformer.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	"sigs.k8s.io/yaml"
+)
+
+// Sort the resources using a customizable ordering based of Kind.
+// Defaults to the ordering of the GVK struct, which puts cluster-wide basic
+// resources with no dependencies (like Namespace, StorageClass, etc.) first,
+// and resources with a high number of dependencies
+// (like ValidatingWebhookConfiguration) last.
+type plugin struct {
+	SortOptions *types.SortOptions `json:"sortOptions,omitempty" yaml:"sortOptions,omitempty"`
+	// TODO: We can't actually be 100% sure that this isn't the top-level
+	// kustomization, as that information is not propagated through
+	// PluginHelpers. Why not? Because propagating this information would
+	// require extensive code changes and changes to the public API, so we opt
+	// to start small. For now, the transformer will only be called explicitly
+	// if the appropriate fields are set in the kustomization.yaml file, which
+	// will also set the IsTopLevel field.
+	//
+	// See: https://github.com/kubernetes-sigs/kustomize/pull/4019#issuecomment-1159771417
+	IsTopLevel bool
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+// Nothing needed for configuration.
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+
+	// If config was called then this is not the root
+	err = yaml.Unmarshal(c, p)
+	if err != nil {
+		return
+	}
+
+	// Default to FIFO sort, aka no-op.
```

Comment:
Everything below needs to happen whether or not we ran Config, which you rightly observe will not happen from (b *Kustomizer) Run. Right now, Run is duplicating the defaulting and not running the validation AFAICT. If you can't call Config from Run, I'd move these out to a method you can call separately from both places, or perhaps move them into the beginning of Transform.